pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -719,3 +595,38 @@ func (exp *flowExporter) findFlowType(conn flowexporter.Connection) uint8 {
 		return 0
 	}
 }
+
+// ValidateAndUpdateStats validates and updates the flow record given the connection
+// key. Caller is expected to grab lock.
+func (exp *flowExporter) ValidateAndUpdateStats(pqItem *flowexporter.ItemToExpire) {
+	// Update the stats in flow record after it is sent successfully
+	pqItem.PrevPackets = pqItem.Conn.OriginalPackets
+	pqItem.PrevBytes = pqItem.Conn.OriginalBytes
+	pqItem.PrevReversePackets = pqItem.Conn.ReversePackets
+	pqItem.PrevReverseBytes = pqItem.Conn.ReverseBytes
+}
+
+func (exp *flowExporter) sendItemToExpire(pqItem *flowexporter.ItemToExpire, isDenyConn bool) error {
+	if pqItem.IsActive {
+		// TODO: more records per data set will be supported when go-ipfix supports size check when adding records
+		if err := exp.addConnToSet(pqItem.Conn, isDenyConn); err != nil {
+			return err
+		}
+		if _, err := exp.sendDataSet(); err != nil {
+			return err
+		}
+		exp.numDataSetsSent = exp.numDataSetsSent + 1
+		if !isDenyConn {
+			klog.V(4).InfoS("Record for connection sent successfully", "flowKey", pqItem.ConnKey, "connection", pqItem.Conn)
+		} else {
+			klog.V(4).InfoS("Record for deny connection sent successfully", "flowKey", pqItem.ConnKey, "connection", pqItem.Conn)
+		}
+	}
+	return nil
+}
+
+func (exp *flowExporter) resetItemToExpire(pqItem *flowexporter.ItemToExpire, currTime time.Time) {
```

Comment:
move this and `ValidateAndUpdateStats` to priority_queue.go? And possibly merge them if not used anywhere else.

Reply:
will do